api/v1beta1/types.go

```diff
@@ -893,6 +906,17 @@ func (s SubnetSpec) IsIPv6Enabled() bool {
 	return false
 }
 
+// GetSecurityRuleByDestination returns security group rule, which matches provided destination ports.
+func (s SubnetSpec) GetSecurityRuleByDestination(ports string) *SecurityRule {
+	for _, rule := range s.SecurityGroup.SecurityRules {
```

Comment:
Probably for the long term, we should change `s.SecurityGroup.SecurityRules` to a dictionary type for faster processing.